src/fields.cpp

```diff
@@ -213,6 +213,44 @@ fields_chunk::~fields_chunk() {
   if (new_s && new_s->refcount-- <= 1) delete new_s; // delete if not shared
 }
 
+static void split_into_tiles(grid_volume gvol, std::vector<grid_volume> &result) {
```

Comment:
As a convention, it would be nice to use pointers for out arguments. Otherwise, it is much harder to reason about the code at the call site while reading it.

Reply:
Fixed.